Modified CONTRIBUTING.md (#2328)
layer5io/layer5: CONTRIBUTING.md

We are beyond excited to see that you want to contribute! We would love to accept your contributions. Layer5 is built by the community and warmly welcomes collaboration. There are many ways in which one could contribute to Layer5 and every contribution is equally appreciated here. Navigate through the following to understand more about contributing here. 

# Before you get started

## Code of Conduct

Layer5 follows the [Cloud Native Computing Foundation (CNCF) Code of Conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md).

Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting community@layer5.io.
Violation of the code of conduct is taken seriously, kindly <a href="https://docs.google.com/forms/d/e/1FAIpQLSeWzC5HjlHugFjB0TtaAVnSkPPqsRQ3JRYjdwyDXf0oyRxcdQ/viewform"> report any violations</a> of the Code of Conduct by filling in the <a href="https://docs.google.com/forms/d/e/1FAIpQLSeWzC5HjlHugFjB0TtaAVnSkPPqsRQ3JRYjdwyDXf0oyRxcdQ/viewform"> Incident report </a>.
The comfort and safety of Layer5 community members is our priority. Please do well to adhere to the Code of Conduct to participate in the Layer5 community.

> ## Style guide

Reviewer: You left the most important part of all xD

```suggestion
## Newcomers guide
To help you get started on contributing to the Layer5 projects, we made a newcomers guide to make contributions easier for new folks like you! Refer the [newcomers guide](https://layer5.io/community/newcomers) for the resources and tutorials to help you get started!

## Style guide
```

Author: I missed it xd

Author: done!

The Layer5 website is hosted in this repository and is built using Gatsbyjs. Before opening a pull request, please review the [design doc](https://docs.google.com/document/d/1rvUZy2_S1a2_14BAQIg6b9cMhUuu04kYzkOPDPaPptI/edit#) to know more about the functioning of the website. Once a pull request has been submitted, a preview deployment will be built and made available to you on your PR.

## Documentation and site discussions 

Join the [#websites](https://layer5io.slack.com/archives/C015QJKUMPU) channel on our slack to discuss about new features, site improvements, enhancement in user experience, and on any other improvements in the documentation.

# Contributing to Layer5's code

If you want to start contributing to the project's code, please follow these guidelines before creating a pull request: 

> - First step is to set up the local development environment. See [this](# How to contribute) on how to do the same. 
> - Bug fixes are always welcome. Start by reviewing the [list of bugs](https://github.com/layer5io/layer5/labels/kind%2Fbug).
> - A good way to easily start contributing is to pick and work on a [good first issue](https://github.com/layer5io/layer5/labels/good%20first%20issue). We try to make these issues as clear as possible and provide basic info on how the code should be changed, and if something is unclear feel free to ask for more information on the issue.
> - We regularly discuss about new issues to work on in our [#websites](https://layer5io.slack.com/archives/C015QJKUMPU) channel on slack. Feel free to join and discuss about any issue or any idea that you may have.

Reviewer: ```suggestion
- First step is to set up the local development environment. See [this](#How to contribute) on how to do the same. 
- Bug fixes are always welcome. Start by reviewing the [list of bugs](https://github.com/layer5io/layer5/labels/kind%2Fbug).
- A good way to easily start contributing is to pick and work on a [good first issue](https://github.com/layer5io/layer5/labels/good%20first%20issue). We try to make these issues as clear as possible and provide basic info on how the code should be changed, and if something is unclear feel free to ask for more information on the issue.
- We regularly discuss about new issues to work on in our [#websites](https://layer5io.slack.com/archives/C015QJKUMPU) channel on slack. Feel free to join and discuss about any issue or any idea that you may have.
```

Author: @alphaX86  Did you suggest that I indent it with tabs? If yes then, I did it now :)

  ```
  git clone https://github.com/layer5io/layer5.git
  ```

  **3.** Navigate to the project directory.

  ```
cd layer5
  ```

  **4.** Always take a pull from the upstream repository to your master branch to keep it at par with the main project(updated repository).

  ```
  git pull upstream master
  ```

  **5.** Create a new branch.

  ```
  git checkout -b <your_branch_name>
  ```

  **6.** Install the dependencies for running the site.

  ```
  make setup-libs
  ```

  **7.** Make the desired changes.

  **8.** Run the site locally to preview changes.

  ```
  make site
  ```

This will run a local web server with "live reload" conveniently enabled. ( **NOTE**: while using make command on Windows, there sometimes arises an error in identifying the command even after it is installed (unrecognized command), this is because the PATH for the binary might not be set correctly ).

  **9.** Track your changes.

  ```
  git add .
  ```

  **10.** Commit your changes. To contribute to this project, you must agree to the [Developer Certificate of Origin (DCO)](## Signing-off on Commits (Developer Certificate of Origin)) for each commit you make.

  ```

>   git commit --signoff -m "<commit subject>"

Reviewer 2: in addition to this command maybe we could give them an option for a relatively shorter command `git commit -s -m "<commit subject>"`

Author: Yup for sure!

Author: done!

  ```

> 

Reviewer 2: maybe before pushing the changes, we could ask them to pull changes in the master from upstream and rebase their branch with the master branch. Something like in the below images : 

![](https://imgur.com/SlKN5vY.png)

![](https://imgur.com/2U1VvhP.png)

>   **11.** Push the committed changes in your feature branch to your remote repo.

Reviewer 2: maybe before pushing the changes, we could ask them to pull changes in the master from upstream and rebase their branch with the master branch. Something like in the below images : 

![](https://imgur.com/SlKN5vY.png)

![](https://imgur.com/2U1VvhP.png)

Author: sounds good.

  ```
  git push -u origin <your_branch_name>
  ```

>   **12.** To create a pull request, click on `compare and pull requests`. Please ensure that you compare your feature branch to the desired branch of the repo you are supposed to make a PR to.

Reviewer 2: ```suggestion
  **12.** Once you’ve committed and pushed all of your changes to GitHub, go to the page for your fork on GitHub, select your development branch, and click the pull request button.. Please ensure that you compare your feature branch to the desired branch of the repo you are supposed to make a PR to.``` If you need to make any adjustments to your pull request, just push the updates to GitHub. Your pull request will automatically track the changes in your development branch and update it.

Author: Nice point!

Author: done!

To signify that you agree to the DCO for contributions, you simply add a line to each of your
git commit messages:

  ```
  Signed-off-by: Jane Smith <jane.smith@example.com>
  ```

In most cases, you can add this signoff to your commit automatically with the
`-s` or `--signoff` flag to `git commit`. You must use your real name and a reachable email
address (sorry, no pseudonyms or anonymous contributions). An example of signing off on a commit:

  ```

>   $ commit -s -m “my commit message w/signoff”

Reviewer: ```suggestion
  $ git commit -s -m “my commit message w/signoff”
```

Author: done!

To signify that you agree to the DCO for contributions, you simply add a line to each of your
git commit messages:

  ```
  Signed-off-by: Jane Smith <jane.smith@example.com>
  ```

In most cases, you can add this signoff to your commit automatically with the
`-s` or `--signoff` flag to `git commit`. You must use your real name and a reachable email
address (sorry, no pseudonyms or anonymous contributions). An example of signing off on a commit:

  ```
  $ git commit -s -m “my commit message w/signoff”
  ```

> To ensure all your commits are signed, you may choose to add this alias to your global ```.gitconfig```:

Author: @alphaX86 @shubham-tyagi @warunicorn19 Can anyone of you please share how you set up your .gitconfig file for the commits cuz I'm having some issues with DCO so thought I'll add these too in the doc so that it is much easier for others too on how to set up commits with the DCO sign up.
I checked some of your commits in other PR's and when you compare it with my commits they look different.

Reviewer: In .gitconfig huh... let me check...

Reviewer: ![image](https://user-images.githubusercontent.com/59508546/140695436-b48c2eca-27cf-4c5f-bafb-2d70def3f16f.png)

This is mine actually and I didn't add any of the things xD Just enter -s that's all

Author: so you mean :
`$ git commit -s -m “New Change”`
That's it?
Cuz I used to do this : 
`$ git commit --signoff -m " Commited suggestions_2  Signed-off-by: Deepak Gupta <deepakgdkg1g8868@gmail.com>"`
and weirdly it doesn't look like how you all commit. It looks like this :
![Screenshot (1082)](https://user-images.githubusercontent.com/81250703/140696497-e2824038-a6c5-424b-9f72-371e53c69b7a.png)

Reviewer: > Cuz I used to do this :
$ git commit --signoff -m " Commited suggestions_2 Signed-off-by: Deepak Gupta <deepakgdkg1g8868@gmail.com>"

Well you don't need to include that Sign-off line in commit message actually :') Git automatically makes one if you give -s or --signoff flag

Author: So @alphaX86, finally simply do this right?
`$ git commit -s -m “New Change”`

this confused me:
`$ commit -s -m “my commit message w/signoff”`

Reviewer 2: Yes, I too just add -s in my git commit command. We don't need to include signoff in the actual commit message.
`git commit -s -m "<commit msg>"` is good enough.

Author: Aight thanks @alphaX86 @shubham-tyagi .

#### Possible badges include:

- <img src="https://github.com/layer5io/layer5/blob/master/src/sections/Community/Member-single/community-green.svg" width="25px" height="25px"/> Community
- <img src="https://github.com/layer5io/layer5/blob/master/src/sections/Community/Member-single/landscape-green.png" width="25px" height="25px"/> Landscape
- <img src="https://github.com/layer5io/layer5/blob/master/src/sections/Community/Member-single/layer5-image-hub.svg" width="25px" height="25px"/> ImageHub
- <img src="https://github.com/layer5io/layer5/blob/master/src/sections/Community/Member-single/meshery-logo-light.svg" width="25px" height="25px"/> Meshery
- <img src="https://github.com/layer5io/layer5/blob/master/src/sections/Community/Member-single/meshery-operator-dark.svg" width="25px" height="25px"/> Meshery Operator
- <img src="https://github.com/layer5io/layer5/blob/master/src/sections/Community/Member-single/smp-dark-text.svg" width="25px" height="25px"/> SMP

> - <img src="https://github.com/layer5io/layer5/blob/master/src/sections/Community/Member-single/nighthawk-logo.png" width="25px" height="25px"/> Nighthawk

Reviewer: ```suggestion
- <img src="https://github.com/layer5io/layer5/blob/master/src/sections/Community/Member-single/getnighthawk-logo.png" width="25px" height="25px"/> Nighthawk
```

Author: Donee
